media_cdn/dualtoken.rb

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env ruby
+#
+# Copyright 2023 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START mediacdn_dualtoken_sign_token]
+require 'base64'
+require 'time'
+require 'openssl'
+require "ed25519"
+
+class Dualtoken
+    def base64_encoder(value)
+        """
+        Returns a base64-encoded string compatible with Media CDN.
+
+        Media CDN uses URL-safe base64 encoding and strips off the padding at the
+        end.
+        """
+        encoded_str = Base64.urlsafe_encode64(value).encode("utf-8")
+        return encoded_str.delete("=")
+    end
+
+    def sign_token(base64_key:, signature_algorithm:, start_time: nil, expiration_time: nil, full_path: nil, path_globs: nil, url_prefix: nil, session_id: nil, data: nil, headers: nil, ip_ranges: nil)
+        """
+        Gets the Signed URL Suffix string for the Media CDN' Short token URL requests.
+        One of (`url_prefix`, `full_path`, `path_globs`) must be included in each input.
+        Args:
+            base64_key: Secret key as a base64 encoded string.
+            signature_algorithm: Algorithm can be either `SHA1` or `SHA256` or `Ed25519`.
+            start_time: Start time as a UTC datetime object.
+            expiration_time: Expiration time as a UTC datetime object. If None, an expiration time 5 mins from now will be used.
```

Comment:
Change the second sentence to: If a value is not specified, the key expires within five minutes.